src/literal.h

```diff
@@ -34,6 +35,7 @@ class Literal {
     int32_t i32;
     int64_t i64;
     uint8_t v128[16];
+    const char* func; // function name for funcref
```

Comment:
There is a [`static_assert`](https://github.com/WebAssembly/binaryen/blob/f2ba91b6340a8c3099ab6a48700c717cbd33599e/src/binaryen-c.cpp#L48-L49) to check if the size of `Literal` and `BinaryenLiteral` are the same. And I wanted to avoid using `Name` in `binaryen-c.h` because this file uses a minimal number of classes from the Binaryen C++ implementation. So I used `const char*` in `BinaryenLiteral`, and to satisfy that `static_assert`, this had to be `const char*` too. (Actually, when I used `Name` in `Literal` and `const char*` in `BinaryenLiteral`, the `static_assert` was OK in Linux, but it failed in Windows. Anyway I think it is safer to use the same set of types in both places.)

Reply:
I'm surprised the size is different, as a Name just contains a `char*`? Perhaps windows alignment works differently for classes with a single element?

What worries me about using a `char*` here is the lifetime of the string. For Names we manage that globally.

If we can't use a `Name`, perhaps that could be an internal detail, that is the constructor would get a Name, and `getFunc` would return a `Name`? (a possible downside for the latter is that it might hash the string, when we know that isn't necessary)

Reply:
Now I've changed the `func` in the `Literal` to `Name` in the union and it seems OK. (`BinaryenLiteral` still has `const char*`.) I'm not 100% sure what the previous problem was, but I guess it was when I didn't include it within the union but made it a separate field in `Literal` (and `BinaryenLiteral`), which was when I first implemented reference types' `Literal` differently. So, in short, nevermind. :)